openedx/core/djangoapps/content_tagging/tasks.py

```diff
@@ -152,3 +153,49 @@ def delete_xblock_tags(usage_key_str: str) -> bool:
     except Exception as e:  # pylint: disable=broad-except
         log.error("Error deleting tags for XBlock with id: %s. %s", usage_key, e)
         return False
+
+
+@shared_task(base=LoggedTask)
+@set_code_owner_attribute
+def update_library_block_tags(usage_key_str: str) -> bool:
+    """
+    Updates the automatically-managed tags for a content library block
+    whenever it is created/updated
+
+    Params:
+        usage_key_str (str): identifier of the Library Block
+    """
+    try:
+        usage_key = LibraryUsageLocatorV2.from_string(usage_key_str)
+
+        log.info("Updating tags for Library Block with id: %s", usage_key)
+
+        # Apply the tag for the default language
+        default_lang_code = settings.LANGUAGE_CODE
```

Comment:
The spec/ticket says: 

> For the MVP, the automatic tag set on block creation should be set to the preferred language of the user who created it (either as set in Open edX or as determined from their browser).

Which is different from `settings.LANGUAGE_CODE`.

Perhaps in `auto_tag_library_block` you can use `crum.get_current_request()` to get the request/user, then you can access the user's preferred language as `request.LANGUAGE_CODE` and pass it to `update_library_block_tags`. (I think `request.LANGUAGE_CODE` should be sufficient already - it comes from the user's profile, "dark lang config", and/or browser via some complex logic which is [already handled by middleware](https://github.com/openedx/edx-platform/blob/d56c2465af958cf26117014e35d7beb73fa41fb3/openedx/core/djangoapps/lang_pref/middleware.py#L26-L57)).

Reply:
Got it, thanks for the heads up! Following your suggestion with `crum.get_current_request()` worked quite well, thanks for that!